### ISSUE TITLE
installation removes other packages in the directory

### ISSUE THREAD

**Issue by ghost** [jmd-dk/Nbody#7]
Hi Jeppe,

I tried bash <(wget -O- https://raw.githubusercontent.com/jmd-dk/concept/${concept_version}/install),
as a result, it removes other packages in the directory where I tried to install concept, if I didn't do something wrong.
I think it is necessary to note that in the documentation. 

Thanks!
Zhejie

**Comment by jmd-dk** [comment]
Thanks for bringing this to my attention. That is indeed unfortunate behavior.

You should install CO*N*CEPT into a dedicated (non-existing or empty) directory. If you choose an existing directory, it does *not* create a sub-directory for the installation within this chosen directory. Unfortunately, the "cleanup" functionality (which allows one to resume the installation process from a previous attempt) sees other files and directories within the chosen installation directory as stuff to throw away, in order to produce a clean installation.

This should indeed be better documented. Better yet, a (improved) check should be in place, resulting in a warning and a `y/n` prompt to the user, if they select a non-dedicated directory.

Sorry for any inconvenience this may have caused!

**Comment by tsfloss** [comment]
This 'unfortunate behavior' just wiped my entire home folder and destroyed several weeks of work.

**Comment by jmd-dk** [comment]
@tsfloss Oh no! :disappointed: 
May I ask how you specified the installation directory? If you do
```bash
bash <(wget -O- https://raw.githubusercontent.com/jmd-dk/concept/v1.0.0/install)
```
and select e.g. `/home`, `/home/<user>`, `$HOME` or `~`, it will tell you that this is a bad idea and suggest a different path.

**Comment by tsfloss** [comment]
yes I got this notification and answered "n", because I wanted to back out anyway. I was not expecting it to proceed and especially not WIPE everything in the folder that it had nothing to with! Please remove or restrict the cleaning behaviour asap...

**Comment by jmd-dk** [comment]
I've [updated](https://github.com/jmd-dk/concept/commit/af3d7b828947e774fbe7b0331da0ba352cb4cfe4) master.
If installing directly into an existing directory instead of a dedicated one, only files/directories with names matching those of CO*N*CEPT files/directories will be removed. That is, doing
```bash
bash <(wget -O- https://raw.githubusercontent.com/jmd-dk/concept/master/install)
```
and selecting `/path/to/concept` as the installation directory, pre-existing files/directories like `/path/to/concept/src`, `/path/to/concept/Makefile`, etc., will be removed (to later be replaced with the content of the CO*N*CEPT [repo](https://github.com/jmd-dk/concept)). Unrecognized files/directories will be left in place.

To be sure, installing into a non-dedicated directory is almost never what one wants. Doing so by mistake is now relatively harmless, except for accidental name clashes as written above.

Also, the safer behavior is only available on master (for now), not v1.0.0.

**Comment by jmd-dk** [comment]
The installation is now not only safe to do in a pre-existing directory, but additionally warns against it. It also now suggests an installation directory, so that it's clear that one should specify the full path. These and a few other fixes are bundled into [release 1.0.1](https://github.com/jmd-dk/concept/releases/tag/v1.0.1).

Thank you both for letting me know about this problem, and once again apologies for the trouble.
